internal/deviations/deviations.go

```diff
@@ -72,20 +74,60 @@ import (
 	"github.com/openconfig/ondatra"
 )
 
+func matchRegex(regex string, deviceInfo string) bool {
+	match, err := regexp.MatchString(regex, deviceInfo)
+	if err != nil {
+		log.Errorf("Error with regex match %v", err)
+	} else if match {
+		return true
+	}
+	return false
+}
+
 // lookupDutDeviations returns the deviations for the specified dut.
 func lookupDUTDeviations(dut *ondatra.DUTDevice) *mpb.Metadata_Deviations {
```

Comment:
we need this to fail if multiple platforms match the dut

Reply:
done